src/HashMap.java

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * Created by david on 4/19/16.
+ */
+public class HashMap implements hash {
+    int tableSize;
+    String[] keyTable;
+    int[] valueTable;
+    int size;
+    float loadFactor = 0.70f;
+    Prime primeNumberGenerator;
+
+    HashMap() {
+        clear();
+    }
+
+    @Override
```

Comment:
What happens if key is null from the caller's point of view? Do they have any way of verifying if the value was put in?